### ISSUE TITLE
Encryption of Media Files

### ISSUE THREAD

**Issue by aimanabsi** [signalapp/libsignal-protocol-javascript#79]
is there a way to encrypt and send media files like pdf,png via this lib?

**Comment by seanieb** [comment]
Not really. How it's normally done is you encrypt the file using a symmetric encryption method, upload the file to a storage service like s3. Then you send the key and the location of the file to the recipient via the signal library.

**Comment by aimanabsi** [comment]
then how can i send pdf file using this api lib ?

**Comment by seanieb** [comment]
If you're just trying trying to send encrypted files I would recommend:

https://github.com/magic-wormhole/magic-wormhole
OR using
https://github.com/FiloSottile/age

**Comment by kitlu007** [comment]
@aimanabsi Hello. Im implementing the libsignal library too in flutter. While text are encrypted, what about media files like images, videos, gifs, docx, pdfs users send toneach other. Arent they encrypted part of the library ? If not , how do i do it ?

**Comment by wez378** [comment]
****
